Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLWriter.java

```diff
@@ -0,0 +1,340 @@
+                            ((NodeReference) object).getIdentifier());
+                    }
+                } else {
+                    bagElements.put(relation, objects);
+                }
+            }
+        }
+
+        if ((first != Node.FIRST_INDEX) && !orderedElements.isEmpty()) {
+            throw new IndexOutOfBoundsException(
+                    "First element at illegal index " + first + ", must be " + Node.FIRST_INDEX);
+        }
+        if (orderedElements.size() != last) {
+            throw new IndexOutOfBoundsException(orderedElements.size() + " elements found, but " + last + " expected.");
+        }
+
+        for (Entry<Long, ObjectType> entry : orderedElements.entrySet()) {
+            ObjectType object = entry.getValue();
+            if (object instanceof Node) {
+                out.appendChild(toElement(object, xml, abbr));
+            } else if (object instanceof Literal) {
+                out.appendChild(xml.createTextNode(((Literal) object).getValue()));
+            } else if (object instanceof NodeReference) {
+                out.appendChild(xml.createTextNode(((NodeReference) object).getIdentifier()));
+            } else {
+                throw new ClassCastException(object.getClass().getName());
+            }
+        }
+
+        for (Entry<String, Collection<ObjectType>> statement : bagElements.entrySet()) {
+            String elementURL = statement.getKey();
+            Element predicate = xml.createElement(abbr.abbreviateElement(RDF.PREDICATE.getIdentifier()));
+            predicate.setAttribute(abbr.abbreviateAttribute(RDF.PREDICATE.getIdentifier(), RDF.VALUE.getIdentifier()),
+                elementURL);
+            predicate.setAttribute(
+                abbr.abbreviateAttribute(RDF.PREDICATE.getIdentifier(), RDF.PROPERTY.getIdentifier()),
+                RDF.BAG.getIdentifier());
+            for (ObjectType object : statement.getValue()) {
+                predicate.appendChild(toElement(object, xml, abbr));
+            }
+            out.appendChild(predicate);
+        }
+    }
+
+    /**
+     * Transforms a node to a DOM document.
+     *
+     * @param node
+     *            node to transform
+     * @param namespaces
+     *            An external Mapping of namespaces to abbreviations. May be
+     *            null.
+     * @return the DOM document
+     */
+    private static Document toDocument(Node node, Map<String, String> namespaces) {
+        Namespaces abbr = namespaces instanceof Namespaces ? (Namespaces) namespaces : new Namespaces(namespaces);
+        DocumentBuilder builder;
+        try {
+            builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        } catch (ParserConfigurationException e) {
+            String message = e.getMessage();
+            throw new RuntimeException(message != null ? message : e.getClass().getSimpleName(), e);
```

Comment:
Is a `RuntimeException` the only one solution in this case? It is hard to catch them and not described in the JavaDoc that this kind of Exception is thrown.

Reply:
This exception will never be thrown. This is just a catch-and-wrap so that the method does not need to declare `throws ParserConfigurationException` which it never does, and the programmer using this method does not have to care for an exception that is never thrown. The definition is that a ParserConfigurationException will be thrown *if a DocumentBuilder cannot be created which satisfies the configuration requested*. Since I fetch a new instance and do not put any configuration before calling `newDocumentBuilder()`, this will never happen.

Reply:
But if this is happen because without a reason this exception is not thrown or declared a mostly not fetched `RuntimeException` is thrown.

Reply:
This is the recommended way to handle exceptions that are “never” thown, see [here](http://www.odi.ch/prog/design/newbies.php#25).